root/static/scripts/common/components/WorkListEntry.js

```diff
@@ -9,42 +9,61 @@
 
 import * as React from 'react';
 
+import {withCatalystContext} from '../../../../context';
+import RatingStars from '../../../../components/RatingStars';
 import loopParity from '../../../../utility/loopParity';
 
 import ArtistCreditLink from './ArtistCreditLink';
 import ArtistRoles from './ArtistRoles';
 import CodeLink from './CodeLink';
+import DynamicAttributesList from './DynamicAttributesList';
 import EntityLink from './EntityLink';
 
 type WorkListRowProps = {|
-  +hasIswcColumn: boolean,
-  +hasMergeColumn: boolean,
+  +$c: CatalystContextT,
+  +checkboxes?: string,
+  +hasAttributesColumn?: boolean,
+  +hasIswcColumn?: boolean,
+  +hasRatingsColumn?: boolean,
+  +seriesItemNumbers?: {+[number]: string},
   +work: WorkT,
 |};
 
 type WorkListEntryProps = {|
-  +hasIswcColumn: boolean,
-  +hasMergeColumn: boolean,
+  +checkboxes?: string,
+  +hasAttributesColumn?: boolean,
+  +hasIswcColumn?: boolean,
+  +hasRatingsColumn?: boolean,
   +index: number,
   +score?: number,
+  +seriesItemNumbers?: {+[number]: string},
   +work: WorkT,
 |};
 
-export const WorkListRow = ({
+export const WorkListRow = withCatalystContext(({
+  $c,
+  checkboxes,
+  hasAttributesColumn,
   hasIswcColumn,
-  hasMergeColumn,
+  hasRatingsColumn,
+  seriesItemNumbers,
   work,
 }: WorkListRowProps) => (
   <>
-    {hasMergeColumn ? (
+    {$c.user_exists && checkboxes ? (
```

Comment:
This generalization also fixes a bug where a logged-out user would be shown the checkbox on ISWC pages.